gui/src/gui.rs

```diff
@@ -298,47 +283,237 @@ impl Application for LedgerInstaller {
+
+    // We do not allow user to click the button if service still processing a task w/ device
+    let genuine_msg = if !device_busy {
+        Some(Message::GenuineCheck)
+    } else {
+        None
     };
-    let mut button = Button::new(
-        Text::new(button_text)
-            .size(11)
-            .width(100)
-            .horizontal_alignment(Horizontal::Center),
-    );
 
-    match (version, genuine_test_running) {
-        (Version::Installed(_), false) => {
-            // button = button.on_press(update_msg);
+    // allow user to check if device is genuine only once per launch
+    let button = if is_genuine.is_none() {
+        Some(Button::new("Check").on_press_maybe(genuine_msg))
+    } else {
+        None
+    };
+
+    let genuine = is_genuine.map(|g| {
+        if g {
+            Text::new(" Yes ")
+        } else {
+            // FIXME: should we display in a more obvious way?
+            Text::new("No!")
         }
-        (Version::NotInstalled, false) => {
-            button = button.on_press(install_msg);
+    });
+
+    let first_column_offset = 80;
+    let first_column_width = 150;
+
+    Container::new(
+        Column::new()
+            .push(
+                Row::new()
+                    .push(Space::with_width(first_column_offset))
+                    .push(Text::new("Model:").width(first_column_width))
+                    .push(Space::with_width(Length::Fill))
+                    .push(Text::new(model))
+                    .push(Space::with_width(Length::Fill)),
+            )
+            .push(Space::with_height(5))
+            .push(
+                Row::new()
+                    .push(Space::with_width(first_column_offset))
+                    .push(Text::new("Firmware:").width(first_column_width))
+                    .push(Space::with_width(Length::Fill))
+                    .push(Text::new(version))
+                    .push(Space::with_width(Length::Fill)),
+            )
+            .push(Space::with_height(5))
+            .push(
+                Row::new()
+                    .push(Space::with_width(first_column_offset))
+                    .push(Text::new("Genuine:").width(first_column_width))
+                    .push(Space::with_width(Length::Fill))
+                    .push_maybe(button)
+                    .push_maybe(genuine)
+                    .push(Space::with_width(Length::Fill)),
+            ),
+    )
+    .style(theme::Container::Frame)
+    .padding(10)
+}
+
+fn apps_container<'a>(
+    bitcoin_version: Version,
+    bitcoin_latest: Version,
+    test_version: Version,
+    test_latest: Version,
+    device_busy: bool,
+) -> Container<'a, Message, Theme, Renderer> {
+    let network_size = 25;
+    let version_color = theme::color::GREY_3;
+    let vertical_rule_position = 230;
+
+    // It looks weird that we load iconex-icons.ttf by its name: Untitled1
+    const ICONEX_ICONS: Font = Font::with_name("Untitled1");
+
```

Comment:
It's not clear to me how Iced handle the font, i guess `Untitled1` is the inner name of the font (that seems not set...) in `iconex-icons.ttf`, but i'm not sure if the font is loaded in our binary or call by OS....